doc/nrf/ug_thread_architectures.rst

```diff
@@ -113,100 +109,91 @@ For more information about the multiprotocol feature, see :ref:`ug_multiprotocol
-|Hardware platforms              |PCA        |Board name                                      |Build target                   |
-+================================+===========+================================================+===============================+
-|:ref:`nRF52840 DK <ug_nrf52>`   |PCA10056   |:ref:`nrf52840dk_nrf52840 <nrf52840dk_nrf52840>`|``nrf52840dk_nrf52840``        |
-+--------------------------------+-----------+------------------------------------------------+-------------------------------+
-|:ref:`nRF52833 DK <ug_nrf52>`   |PCA10010   |:ref:`nrf52833dk_nrf52833 <nrf52833dk_nrf52833>`|``nrf52833dk_nrf52833``        |
-+--------------------------------+-----------+------------------------------------------------+-------------------------------+
-|:ref:`nRF5340 DK <ug_nrf5340>`  |PCA10095   |:ref:`nrf5340dk_nrf5340 <nrf5340dk_nrf5340>`    |``nrf5340dk_nrf5340_cpuapp``   |
-|                                |           |                                                |                               |
-|                                |           |                                                |``nrf5340dk_nrf5340_cpuappns`` |
-+--------------------------------+-----------+------------------------------------------------+-------------------------------+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuappns
 
 .. _thread_architectures_designs_cp:
 
 Co-processor designs
 ********************
 
-In the co-processor designs, with either network co-processor (NCP) or radio co-processor (RCP), the application layer runs on a host processor and communicates with OpenThread through a serial connection using a standardized host-controller protocol (Spinel).
-OpenThread can run on either the radio or the host processor.
+In co-processor designs, the application runs on one processor (the host processor) and communicates with another processor that provides the Thread radio.
+The communication happens through a serial connection using a standardized host-controller protocol (Spinel).
+
+OpenThread runs on either the radio processor or the host processor, depending on whether a network co-processor (NCP) design or a radio co-processor (RCP) design is chosen.
 
 .. _thread_architectures_designs_cp_ncp:
 
-Network Co-Processor (NCP)
+Network co-processor (NCP)
 ==========================
 
-The standard NCP design has Thread features on the SoC and runs the application layer on a host processor, which is typically more capable than the OpenThread device, although it has greater power demands.
-The host processor communicates with the OpenThread device through a serial interface (typically UART or SPI) over the Spinel protocol.
+In the standard NCP design, the network-related features run on the processor that provides the Thread radio, and the application layer runs on a host processor.
```

Comment:
Could we mention that the full OpenThread stack runs on the co-processor in this case?